src/phpDocumentor/Descriptor/Traits/HasPackage.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @link https://phpdoc.org
+ */
+
+namespace phpDocumentor\Descriptor\Traits;
+
+use phpDocumentor\Descriptor\Interfaces\ClassInterface;
+use phpDocumentor\Descriptor\Interfaces\ElementInterface;
+use phpDocumentor\Descriptor\Interfaces\InheritsFromElement;
+use phpDocumentor\Descriptor\Interfaces\InterfaceInterface;
+use phpDocumentor\Descriptor\Interfaces\PackageInterface;
+
+trait HasPackage
+{
+    /** @var PackageInterface|string $package The package with which this element is associated */
+    protected $package;
+
+    /**
+     * Sets the name of the package to which this element belongs.
+     *
+     * @internal should not be called by any other class than the assamblers
+     *
+     * @param PackageInterface|string $package
+     */
+    public function setPackage($package): void
+    {
+        $this->package = $package;
+
+        if ($this instanceof ClassInterface || $this instanceof InterfaceInterface) {
+            foreach ($this->getConstants() as $constant) {
+                /** @psalm-suppress UndefinedMethod */
+                $constant->setPackage($package);
+            }
+
+            foreach ($this->getMethods() as $method) {
+                /** @psalm-suppress UndefinedMethod */
+                $method->setPackage($package);
+            }
+        }
+
+        // @phpcs:ignore SlevomatCodingStandard.ControlStructures.EarlyExit.EarlyExitNotUsed
+        if ($this instanceof ClassInterface) {
```

Comment:
Looks like traits are missing here.

Reply:
I was wondering the same thing when I copied this code..